lamindb/_curate.py

```diff
@@ -146,7 +149,7 @@ def non_validated(self) -> list:
         return self._non_validated
 
     @property
-    def fields(self) -> dict:
+    def fields(self) -> Mapping:
```

Comment:
This returns a Python dict.

My understanding is that you'd use `Mapping` when you're typing a param that accepts/returns an object that's dict-like; e.g., a `pd.Series` is dict-like.

But this is not the case here, is it?

If it's not, I'd keep `dict` for simplicity & correctness.

Reply:
Hmm yeah. I think that mypy isn't smart enough to understand that. 

```
lamindb/_curate.py:154: error: Incompatible return value type (got "Mapping[str, Any]", expected "dict[Any, Any]")  [return-value]
lamindb/_curate.py:411: error: Incompatible return value type (got "Mapping[str, Any] | None", expected "dict[Any, Any]")  [return-value]
```

Reply:
It returns whatever `categoricals` was set to. So not always a dict - only when a dict was passed.

Reply:
If this is the code design (which is OK), then we should continue the discussion in the top thread.